ejb3/src/main/java/org/jboss/as/ejb3/tx/OwnableReentrantLock.java

```diff
@@ -115,8 +120,9 @@ public boolean tryLock(long timeValue, TimeUnit timeUnit, Object owner) {
     }
 
     public void unlock(Object owner) {
+        Assert.checkNotNullParam("owner", owner);
         synchronized (this.lock) {
-            if (owner != this.owner) {
+            if (!owner.equals(this.owner)) {
```

Comment:
You'll want to use Objects.equals(...) here as well, for the same reason as above.